src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,71 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { FilterTodos } from '../../types/FilterTodos';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string,
+  onQueryChange: (val: string) => void,
+  selectedFilter: string,
+  onSelectedFilter: (val: FilterTodos) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  onQueryChange,
+  selectedFilter,
+  onSelectedFilter,
+}) => {
+  const resetAllFilters = () => {
+    onQueryChange('');
+    onSelectedFilter(FilterTodos.All);
+  };
```

Comment:
From task:
> the x button should clear the query and reset the todos;

There is no need to set selectedFilter to All after clicking the x